spec/asyncBehaviors.js

```diff
@@ -612,3 +612,180 @@ describe('Rate-limited', function() {
+        });
+
+        it('Set to false initially, should maintain synchronous notification', function() {
+            var observable = ko.observable().extend({deferred:false});
+            var notifySpy = jasmine.createSpy('notifySpy');
+            observable.subscribe(notifySpy);
+
+            observable('A');
+            expect(notifySpy.argsForCall).toEqual([ ['A'] ]);
+        });
+
+        it('Should suppress notification when value is changed/reverted', function() {
+            var observable = ko.observable('original').extend({deferred:true});
+            var notifySpy = jasmine.createSpy('notifySpy');
+            observable.subscribe(notifySpy);
+
+            observable('new');
+            expect(observable()).toEqual('new');
+            observable('original');
+
+            jasmine.Clock.tick(1);
+            expect(notifySpy).not.toHaveBeenCalled();
+            expect(observable()).toEqual('original');
+        });
+
+        it('Set to false, should turn off deferred notification if already turned on', function() {
+            var observable = ko.observable().extend({deferred:true});
+            var notifySpy = jasmine.createSpy('notifySpy');
+            observable.subscribe(notifySpy);
+
+            // First, notifications are deferred
+            observable('A');
+            expect(notifySpy).not.toHaveBeenCalled();
+            jasmine.Clock.tick(1);
+            expect(notifySpy.argsForCall).toEqual([ ['A'] ]);
+
+            notifySpy.reset();
+            observable.extend({deferred:false});
+
+            // Now, they are synchronous
+            observable('B');
+            expect(notifySpy.argsForCall).toEqual([ ['B'] ]);
+        });
+    });
+
+    describe('Observable Array change tracking', function() {
+        it('Should provide correct changelist when multiple updates are merged into one notification', function() {
+            var myArray = ko.observableArray(['Alpha', 'Beta']).extend({deferred:true}),
+                changelist;
+
+            myArray.subscribe(function(changes) {
+                changelist = changes;
+            }, null, 'arrayChange');
+
+            myArray.push('Gamma');
+            myArray.push('Delta');
+            jasmine.Clock.tick(1);
+            expect(changelist).toEqual([
+                { status : 'added', value : 'Gamma', index : 2 },
+                { status : 'added', value : 'Delta', index : 3 }
+            ]);
+
+            changelist = undefined;
+            myArray.shift();
+            myArray.shift();
+            jasmine.Clock.tick(1);
+            expect(changelist).toEqual([
+                { status : 'deleted', value : 'Alpha', index : 0 },
+                { status : 'deleted', value : 'Beta', index : 1 }
+            ]);
+
+            changelist = undefined;
+            myArray.push('Epsilon');
+            myArray.pop();
+            jasmine.Clock.tick(1);
+            expect(changelist).toEqualOneOf([[], undefined]);
```

Comment:
I copied this from my plugin, where it think it was meant to pass with different versions of Knockout. The correct result should be `undefined`.